packages/components/src/modal/header.js

```diff
@@ -18,7 +18,7 @@ const ModalHeader = ( { icon, title, onClose, closeLabel, headingId } ) => {
 			<div className="components-modal__header-heading-container">
 				{ icon &&
 					<span className="components-modal__icon-container" aria-hidden>
-						{ icon }
+						{ icon && <img alt={ __( 'User avatar' ) } src={ icon } /> }
```

Comment:
It looks like this is specific to the Post Locking behavior and shouldn't be baked into the generic Modal component. Can't we just pass the entire `<img />` as `icon` prop instead?

Reply:
Yes, unless the icon should always have an alt tag for accessibility reasons, maybe it makes more sense to add and require an `iconAlt` attribute?